common/src/util/config.ts

```diff
@@ -1,203 +1,301 @@
-import _config from "config";
+import dotenv from "dotenv";
 import {existsSync, readFileSync} from "fs";
+import path from "path";
+
+// ======================== HELPERS ========================
+
+const readSecretFile = (filepath: string): string => readFileSync(filepath).toString()
+    .trim();
+
+const MONOREPO_ROOT = `${__dirname}/../../..`;
+const MONOREPO_CONFIG = `${MONOREPO_ROOT}/config`;
+const MONOREPO_ENV = `${MONOREPO_ROOT}/.env`;
+const MICROSERVICE_CONFIG = "./config";
+
+// ======================== SETUP ==========================
+
+// Ensure that we are loading from root config, as well as local microservice config
+// This setup must happen before the "config" package is imported to have an effect
+process.env.NODE_CONFIG_DIR = [MICROSERVICE_CONFIG, MONOREPO_CONFIG].join(path.delimiter);
+dotenv.config({path: MONOREPO_ENV});
+import _config from "config";
+
+// ======================== CONFIG =========================
 
 export const config = {
     auth: {
         google: {
             get clientId(): string {
-                return readFileSync("./secret/googleClientId.txt").toString();
+                if (_config.has("auth.google.clientId")) {
+                    return _config.get("auth.google.clientId");
+                }
+                return readSecretFile("./secret/googleClientId.txt");
             },
             get secret(): string {
-                return readFileSync("./secret/googleSecret.txt").toString();
+                if (_config.has("auth.google.secret")) {
+                    return _config.get("auth.google.secret");
+                }
+                return readSecretFile("./secret/googleSecret.txt");
             },
             get callbackUrl(): string {
-                return _config.get<string>("auth.google.callbackUrl");
+                return _config.get("auth.google.callbackUrl");
```

Comment:
This shows that we are, at minimum, forcing the user to have a config json file. Is that the design? Makes sense, one json for non-secret stuff and one env for the secrets.